core/src/main/scala/org/apache/spark/deploy/master/MasterSource.scala

```diff
@@ -42,6 +42,6 @@ private[spark] class MasterSource(val master: Master) extends Source {
 
   // Gauge for waiting application numbers in cluster
   metricRegistry.register(MetricRegistry.name("waitingApps"), new Gauge[Int] {
-    override def getValue: Int = master.waitingApps.size
+    override def getValue: Int = master.apps.filter(_.state == ApplicationState.WAITING).size
```

Comment:
"waitingApps" suppose to refer to the number of waiting applications in the Master, and `apps` includes all the uncompleted applications including `waitingApps`. If there's any inconsistent meaning, I think we should find out what actual `waitingApps` usage in `Master`, to see if there's any bug in it.